cmd/svcat/parameters/parameters.go

```diff
@@ -56,19 +57,45 @@ func ParseVariableAssignments(params []string) (map[string]interface{}, error) {
 		}
 		value := strings.TrimSpace(parts[1])
 
+		if strings.ContainsAny(variable, ".") {
+			//check if params form is -p xxx.xx
+			r, _ := regexp.Compile(`[^\s\.]+\.[^\s\.]+(\.[^\s\.]+)*`)
+			if r.MatchString(variable) == true {
+				dotKeys = strings.Split(variable, ".")
+				variable = dotKeys[0]
+			} else {
+				return nil, fmt.Errorf("invalid parameter (%s), must be in x.x format", variable)
+			}
+		}
+
 		storedValue, ok := variables[variable]
 		// Logic to add new value to map variables:
-		// if variable DNE: add pair to variables as variable:value
-		// if variable exists in form of variable:value, create array to hold old value&new value
-		// if variable exists in form variable:[some values], append new value to existing array
+		// if variable DNE and not exploding params -> variable:value
+		// if variable DNE and exploding params -> variable: map[somekey:somevalue]
+		// Logic to add value to variables where variable already exists in variables:
+		// if variable:value -> variable:[old value, newvalue]
+		// if variable:[value1, value2] -> variable:[value1, value2, newvalue]
+
 		if !ok {
-			variables[variable] = value // if there is no key, add key&value as string
+			if len(dotKeys) == 0 {
+				variables[variable] = value
+			} else {
+				for i := len(dotKeys[1:]); i > 0; i-- {
+					if i == len(dotKeys[1:]) {
```

Comment:
OMG, it took me a couple minutes but I see what you are doing now, saving it from the bottom up in nested maps! 😊  👍

Reply:
Yes, the last element in the dot params always has the value "value", so I made that the initial  map and went through the rest of the params backwards to build the nested maps around it. Looking at it now, I think my if-else is unnecessary since it only will fall into the if on the first go-round. I will just do the assignment on the last element and then loop through the rest to build the nested map!